consensus/spos/bls/blsSubroundsFactory.go

```diff
@@ -189,10 +189,15 @@ func (fct *factory) generateBlockSubround() error {
 		return err
 	}
 
-	fct.worker.AddReceivedMessageCall(MtBlockBodyAndHeader, subroundBlock.receivedBlockBodyAndHeader)
-	fct.worker.AddReceivedMessageCall(MtBlockBody, subroundBlock.receivedBlockBody)
-	fct.worker.AddReceivedMessageCall(MtBlockHeader, subroundBlock.receivedBlockHeader)
-	fct.consensusCore.Chronology().AddSubround(subroundBlock)
+	sideChainSubroundBlock, err := NewSideChainSubroundBlock(subroundBlock)
```

Comment:
do we really need this ? this is needed only on the new type of consensus - not in the current one.

Reply:
add a TODO and make the factory on another PR.

Reply:
implemented factory for sovereign chain in this PR